File: GomokuServer/src/GomokuServer.Application/Games/Responses/GetGameHistoryResponse.cs
namespace GomokuServer.Application.Games.Responses;

public class GetGameHistoryResponse
{
	[Required]
	public required int BoardSize { get; init; }

	[Required]
	public required string Gen { get; init; }

	[Required]
	public required int MovesCount { get; init; }

	[Required]
	public required PlayersDto Players { get; init; }

	[Required]
	public required bool IsGameStarted { get; set; }

	[Required]
	public required bool HasBothPlayersJoined { get; set; }

	[Required]
	public required bool IsCompleted { get; set; }

	public string? Winner { get; init; }

	public List<TileDto>? WinningSequence { get; init; }

	[Required]
	public required IReadOnlyDictionary<int, TileDto> MovesHistory { get; init; }

	public TimeControlDto? TimeControl { get; init; }

	public ClockDto? Clock { get; init; }
}

File: GomokuClient/packages/gomoku-api/client/types/GetGameHistoryResponse.ts
import type { ClockDto } from "./ClockDto.ts";
import type { PlayersDto } from "./PlayersDto.ts";
import type { TileDto } from "./TileDto.ts";
import type { TimeControlDto } from "./TimeControlDto.ts";

 export type GetGameHistoryResponse = {
    /**
     * @type integer, int32
    */
    boardSize: number;
    /**
     * @type string
    */
    gen: string;
    /**
     * @type integer, int32
    */
    movesCount: number;
    /**
     * @type object
    */
    players: PlayersDto;
    /**
     * @type boolean
    */
    isGameStarted: boolean;
    /**
     * @type boolean
    */
    hasBothPlayersJoined: boolean;
    /**
     * @type boolean
    */
    isCompleted: boolean;
    /**
     * @type string
    */
    winner?: string | null;
    /**
     * @type array
    */
    winningSequence?: TileDto[] | null;
    /**
     * @type object
    */
    movesHistory: {
        [key: string]: TileDto;
    };
    /**
     * @type object | undefined
    */
    timeControl?: TimeControlDto;
    /**
     * @type object | undefined
    */
    clock?: ClockDto;
};